lib/core/src/irods_client_negotiation.cpp

```diff
@@ -58,10 +58,12 @@ namespace irods
                     continue;
                 }
 
-                const auto hostname_in_provider_hosts_list = std::any_of(
-                    std::cbegin(*provider_hosts), std::cend(*provider_hosts), [&_host_name](const auto& _host) {
-                        return _host_name == _host;
-                    });
+                const auto hostname_in_provider_hosts_list =
+                    std::any_of(std::cbegin(*provider_hosts),
+                                std::cend(*provider_hosts),
+                                [&_host_name](const nlohmann::json& _host) {
```

Comment:
interesting you had to leave `auto`... and this is the only place this was required?   why was it required?

Reply:
I left `auto` because the change wasn't necessary, not because it was required.

Reply:
okay